lib/domain/entities/qr_code_entity.dart

```diff
@@ -0,0 +1,19 @@
+class QRCodeEntity {
+  final String seed;
+  final DateTime? expiresAt;
+
+  QRCodeEntity({
+    required this.seed,
+    this.expiresAt,
+  });
+
+  @override
+  bool operator ==(covariant QRCodeEntity other) {
```

Comment:
**question**: why did you decide to make this `covariant`? 🙂 The superclass does not introduce a covariant parameter and the standard pattern for equality goes like:

```dart
if (identical(this, other) return true;

if (other is ABC) {
  return a == other.a && ...;
}

return false;
```

If you made this `covariant` then you would disallow checks and comparisons with other classes, such as `Object`. Even if this isn't the case, because a `QRCodeEntity` is never equal to `Object`, this method is internally used by `Map` and its implementations. I think you shouldn't go this way 🙂

Reply:
To be very honest with you I used a VSCode extension to create the equality method at the domain layer. I did not notice the Covariant variable. The extension helps me to create this equality because there are a lot of boilerplates when creating it. But you got a point, next time I'll pay more attention to that. Thanks!